platform/cog-platform-drm.c

```diff
@@ -236,6 +279,180 @@ init_egl ()
 }
 
 
+static void
+input_handle_key_event (struct libinput_event_keyboard *key_event)
+{
+    struct wpe_input_xkb_context *default_context = wpe_input_xkb_context_get_default ();
+    struct xkb_state *state = wpe_input_xkb_context_get_state (default_context);
+
+    uint32_t key = libinput_event_keyboard_get_key (key_event) + 8;
+    uint32_t keysym = xkb_state_key_get_one_sym (state, key);
+    uint32_t unicode = xkb_state_key_get_utf32 (state, key);
+
+    struct wpe_input_keyboard_event event;
+    event.time = libinput_event_keyboard_get_time (key_event);
+    event.key_code = keysym;
+    event.hardware_key_code = unicode;
```

Comment:
Also, this seems odd... Hardware key codes almost never are Unicode code points, aren't they?

Reply:
I don't remember exactly how this came to be, but the same is done in the -fdo plugin and elsewhere.